epigraphhub/data/worldbank.py

```diff
@@ -0,0 +1,257 @@
+
+        for source in wb.source.list():
+
+            if keyword in source["name"].replace(" ", "").lower():
+
+                results.append(source)
+
+    return pd.DataFrame.from_records(results)
+
+
+def search_in_indicators(keyword, db=2):
+
+    """
+    Returns a dataframe with the indicators matched by partial name.
+
+    :params keyword: string. keyword to search in the indicators name.
+    :params db:int. Number associated with the database whose you want to get the list
+                    of indicators. You can discover this number in the function 'search_in
+                    _database'. By default the indicators are search over the World Development
+                    Indicators database (db = 2).
+
+    :returns: DataFrame
+
+    """
+
+    results = []
+
+    for indicator in wb.series.list(q=keyword, db=db):
+        results.append(indicator)
+
+    return pd.DataFrame.from_records(results)
+
+
+def get_worldbank_data(ind, country, db=2, time="all", columns=None):
+
+    """
+    This function get a list of indicators according to some country from the world data bank
+    and return this series in a dataframe.
+
+    :params ind: List of strings. List with the indicators whose data you want to get.
+
+    :params country: List of strings|string. List with the ISO-CODE for the countries whose
+                     data you want to get.
+
+    :params time: Interval of  years. If filled `time = 'all'`, the function will
+                  return all the data available. You can also specify a range of years.
+                  For example, if you want to get the data for the period between the
+                  years 2010 and 2020, you can fill this parameter with
+                  `time = range(2010,2021)`.
+
+    :params columns: List of strings. List with names to rename the columns instead of use the
+                    names in the `ind` list.
+
+    Important: The lists `ind` and `columns` must have the same lenght.
+
+    :returns: DataFrame
+
+    """
+
+    if columns != None:
+
+        if len(columns) == len(ind):
+
+            rename_columns = {}
+
+            for i in np.arange(0, len(ind)):
+                rename_columns[ind[i]] = columns[i]
+
+    else:
+
+        rename_columns = {}
+
+    if len(ind) == 1:
+
+        df = wb.data.DataFrame(
+            series=ind, economy=country, db=db, index=["time"], labels=True, time=time
+        )
+
+        df = df.reset_index()
+
+        del df["time"]
+
+        df["Time"] = df["Time"].astype("str")
+
+    else:
+        df = wb.data.DataFrame(
+            series=ind,
+            economy=country,
+            db=db,
+            index=["time", "economy"],
+            labels=True,
+            time=time,
+        )
+
+    if len(rename_columns) != 0:
+
+        df = df.rename(columns=rename_columns)
+
+    new_dates = []
+    periodicity = []
+
+    for i in df["Time"]:
+
+        if len(i) == 4:
+
+            new_dates.append(f"{i}-01-01")
```

Comment:
Are the dates so messed up that they need to be transformed like this?
We definitely need unit tests to make sure this **heuristic** will work for different indicators.

Reply:
Well, I was not able to look at all the databases available on this platform. But, for the majority of data that I looked at, the indicator provides annual data. This annual data should work well with this function. 

I only needed to write this if loop, because in the database  = 15 (`Global Economic Monitor`) I found some monthly data referred to as: '2016M01, 2016M02, ...' and quarterly data referred to as: '2016Q1, 2016Q2, ...'. So I created this loop to take care of this format of data.

Reply:
pandas can deal with these strings
```python
In [5]: pd.to_datetime('2002Q2')
Out[5]: Timestamp('2002-04-01 00:00:00')
# for the months you have to pass a format string.
In [10]: pd.to_datetime('2002M02', format='%YM%m')
Out[10]: Timestamp('2002-02-01 00:00:00')
```

Reply:
with unit tests, you can collect examples and make sure they work with your parse strategy.